src/MatomoTracker.js

```diff
@@ -158,11 +158,23 @@ class MatomoTracker {
       }).toString()
     };
 
-    fetch(this.trackerUrl, fetchObj)
-      .catch((err) => console.error('error', err))
-      .finally(
-        () => this.log && console.log('Matomo tracking is sent:', this.trackerUrl, fetchObj)
-      );
+    return fetch(this.trackerUrl, fetchObj)
+      .then((response) => {
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+
+        this.log && console.log('Matomo tracking is sent:', this.trackerUrl, fetchObj);
+
+        return response;
+      })
+      .catch((error) => {
+        this.log && console.log('Matomo tracking is not sent:', this.trackerUrl, fetchObj);
+
+        console.warn('Matomo tracking error:', error);
+
+        return { error, ok: false, status: 500 };
```

Comment:
Does the response not have an own code that we can pass through?

If it does, we should use that instead, should we not?

Reply:
i haven't found one in the error returned in the catch. the response at success has one, as well as the `ok`. so i wanted to be sure, that this is also the case on errors.

Reply:
does the response not have a status, when `!response.ok`? because if it does and it is the error status code, then we should pass that into the error instead of the hard coded `500`

Reply:
when `!response.ok` there is a `response.status`, yes. would be nice to not have it hardcoded.

Reply:
then please change it :)

Reply:
we haven't found a good way to always return a status code on error.
removed the hardcoded `ok` and `status` and return just the `error`.